nvbench/detail/measure_hot.cu

```diff
@@ -37,7 +37,13 @@ namespace nvbench::detail
 
 measure_hot_base::measure_hot_base(state &exec_state)
     : m_state{exec_state}
-    , m_launch{m_state.get_cuda_stream()}
+    , m_launch{nvbench::launch([this]() -> decltype(auto) {
+      if (!m_state.get_cuda_stream().has_value())
+      {
+        m_state.set_cuda_stream(nvbench::cuda_stream{m_state.get_device()});
+      }
+      return m_state.get_cuda_stream().value();
```

Comment:
This feels weird to have the initialization of the `optional` external to `state`.

How about putting this logic inside `state::get_cuda_stream` instead and don't expose the `optional` externally.

Reply:
> How about putting this logic inside `state::get_cuda_stream` instead and don't expose the `optional` externally.

@allisonvacanti and I have discussed that option too but agreed to prefer explicitly setting the stream over implicitly initializing it as a byproduct, if it didn't exist. Considering the user interfacing with the API, I feel that, for multi-GPU systems, it's safer to make it explicit when resources are created and what device they are associated with. Especially, when the `current device` may influence what device a resource is associated with.

That said, I'm fine to have it whatever we decide makes more sense. @allisonvacanti what do you think?